core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1358,3 +1387,12 @@ private case class OptionAssigner(
     deployMode: Int,
     clOption: String = null,
     confKey: String = null)
+
+private[spark] trait SparkSubmitOperationsClient {
+
+  def kill(args: SparkSubmitArguments): Unit
```

Comment:
initially I had SparkConf but  I see the operations as an extension to to SparkSubmit and initially SparkSubmitArguments is all you have at the point where they are called. It is possible to create a SparkConf from that argument map but then I will need to filter out what is irrelevant, otherwise the structure does not matter but the content matters.

Reply:
> it is possible to create a SparkConf from that argument map but then I will need to filter out what is irrelevant,

Why would you filter out anything? We don't filter out Mesos-related configs when submitting to YARN, for example. YARN just never looks at them.

Same thing here.

I just don't think `SparkSubmitArguments` is a good interface to expose to code outside of `SparkSubmit`.

Reply:
Ok my thinking was that code called is part of the sparksubmit process, for example whatever logging capabilities exist (printstream) should be also inherited. Anyway I can change it.

Reply:
Btw I will change that but I will have to pass `args.submissionToRequestStatusFor` which is suboptimal.

Reply:
I don't see that as sub-optimal, I see that as the goal. e.g. you're providing what to kill as an argument to the method.

Even better if all that could be encoded in a URI, but the master URI syntax was not really designed that way...